### ISSUE TITLE
Error for `KubeletConfiguration` that isn's supported for an AMIFamily

### ISSUE THREAD

**Issue by jonathan-innis** [aws/karpenter#3722]
### Tell us about your request

There are some pieces of the `v1alpha5.KubeletConfiguration` that aren't supported by Bottlerocket AMIFamily. Rather than silently ignoring these pieces, we should error to the user that we can't launch when there are pieces of the kubeletConfiguration that aren't supported by the AMIFamily

### Tell us about the problem you're trying to solve. What are you trying to do, and why is it hard?

N/A

### Are you currently working around this issue?

Silently ignoring `v1alpha5.KubeletConfiguration` that isn't supported by the AMIFamily

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Comment by jonathan-innis** [comment]
Currently, the fields that aren't supported by Bottlerocket but are supported by the Karpenter `kubeletConfiguration` are: 
1. `evictionSoft`
2. `podsPerCore`
3. `cpuCFSQuota`

**Comment by srinchow** [comment]
@jonathan-innis i can take this up if this needs to be done

**Comment by stmcginnis** [comment]
Small update on the unsupported fields listed above.

`settings.kubernetes.cpu-cfs-quota-enforced` and `settings.kubernetes.eviction-soft*` settings are available as of Bottlerocket 1.14.0 and later. 1.14.0 was released in May 11, so hopefully any new deployments would be using a release that includes these settings.

**Comment by ecerulm** [comment]
Related: Add Bottlerocket support for kubeletConfiguration.evictionSoft #4271 
Related: chore: add bottlerocket support for kubeletconfiguration evictionsoft #6760
Related: chore: Add soft eviction setting to Bottlerocket kublet config #6390

**Comment by diranged** [comment]
Agreed btw ... I spent a few hours last night trying to understand why settings were not being applied, until I went digging through the code. Thanks for beginning work on this.
